madl/madl-ferma/src/main/java/com/gentics/madl/frame/AbstractVertexFrame.java

```diff
@@ -146,12 +154,30 @@ public void setSingleLinkOutTo(VertexFrame vertex, String... labels) {
 		return result;
 	}
 
+	@Override
+	public Stream<Edge> streamOutE(String label) {
+		Iterator<Edge> it = getElement().getEdges(Direction.OUT, label).iterator();
+		Stream<Edge> stream = StreamSupport.stream(
+			Spliterators.spliteratorUnknownSize(it, Spliterator.ORDERED),
+			false);
+		return stream;
+	}
+
 	@Override
 	public <T extends ElementFrame> TraversalResult<? extends T> in(String label, Class<T> clazz) {
 		TraversalResult<? extends T> result = new TraversalResult<>(in(label).frameExplicit(clazz));
 		return result;
 	}
 
+	@Override
+	public Stream<Edge> streamInE(String label) {
+		Iterator<Edge> it = getElement().getEdges(Direction.IN, label).iterator();
```

Comment:
Create a private function which takes the direction as an argument to avoid code duplication. Maybe even consider making it public.

Reply:
Yes, I was wondering myself. I try to add those methods to the interface via default methods. This way all implementations can use those generic methods which make use of `getElement`.
WDYT

Reply:
@philippguertler  done

Reply:
Use the new function for streamOutE as well

Reply:
Done